### ISSUE TITLE
Windows 11 compatibility issues

### ISSUE THREAD

**Issue by DirkTeulings** [fredemmott/StreamDeck-AudioSwitcher#58]
**Describe the bug**
When upgraded to windows 11 the plugin is not able to set the default audio device.
Button do turn on and give "color"

**To Reproduce**
1. upgrade to windows 11

**Expected behavior**
Should switch between defaults

**Operating System**
Windows 11 Professional, 64-bit

**Notes**
Willing to test if updates are available.
Also what library are you using for the switching? (Not familiar with c++)

**Comment by izzy** [comment]
Can confirm this is the case after upgrading to Win 11. This plugin has become more essential to me in W11 since the taskbar basically removed the possibility to quickly set the output device via the taskbar.

**Comment by fredemmott** [comment]
From #55 and #57, this appears to vary by device, and for some is fixed by installing windows updates; I suspect this varies by device/driver, but there does not appear to be a consistent issue here. I also do not own any hardware compatible with Windows 11.

> Also what library are you using for the switching? (Not familiar with c++)

https://github.com/fredemmott/audiodevicelib

**Comment by fredemmott** [comment]
Also, try removing the actions and re-adding them to the streamdeck: that will resolve the issue if the upgrade changed the windows device IDs.

**Comment by izzy** [comment]
> Also, try removing the actions and re-adding them to the streamdeck: that will resolve the issue if the upgrade changed the windows device IDs.

That was it!

Usually before if Windows messed up the IDs it was enough to just change to another one and then to the one I want, now it only worked again after removing the action and then readding it.

Is there maybe a way to test if the device still exists/is plugged in and then display a different icon so it's visually clear something went wrong?

**Comment by fredemmott** [comment]
> Is there maybe a way to test if the device still exists/is plugged in and then display a different icon so it's visually clear something went wrong?

Not at the moment. There's a few potential options, all with downsides:
- truly add a custom image. This isn't great as Elgato's software only supports two icons per action. All 3 images would need to be edited through the plugin options rather than the normal action icon settings. The icon settings would still be there, but do nothing
- add an image selection in the plugin options just for this. It adds a fair bit of complexity, and doesn't really feel worth it.
- display the ⚠️ for as long the device is not attached, rather than just on load, on unplug, or when pressed. The problem here is bugs in Elgato's software:
  - you can only show it for a few seconds at a time (not a bug), so the plugin needs to keep showing it
  - regardless of what else happens, when it stop showing the warning triangle, the streamdeck reverts to displaying what it was showing before, ignoring whatever happened since. For example:
     - start: device is inactive, button shows inactive
     - 1 second later: device is unplugged, warning triangle displayed
     - 1 second later: device is plugged in, device becomes active. Plugin tells streamdeck to switch to 'active' state
     - 1 second later: warning triangle times out. streamdeck is now in 'active' state, but incorrectly displays the icon for 'inactive'
     
 I'd like to take the third option, but I'm hoping that/waiting for Elgato to fix the bug.

**Comment by grahamcropley** [comment]
Hopefully not hijacking this issue. But this has stopped working for me after upgrading to Windows 11.

- I have noticed the hardware ID issue, and it showing duplicates, but reselecting the 'correct' ones from the dropdown list removes the duplicates. but still doesn't work.
- I have removed and readded from the deck, and re-setup, but still doesn't work.
- I have also uninstalled and reinstalled the plugin, but still doesn't work.

Toggle Audio Device & Set Audio Device both do nothing.

Are there any troubleshooting or debug steps I can do to help?

Windows 11 - 21H1 22000.282
Stream Deck - 5.1.3.14750
AudioSwitcher - v3
